src/task/RunResharperCodeAnalysisTool.ps1

```diff
@@ -138,6 +139,10 @@ if($filteredElements.Count -eq 0) {
     $summaryMessage = "No code quality issues found!"
     Write-Output ("##vso[task.complete result=Succeeded;]{0}" -f $summaryMessage)
     Add-Content $summaryFilePath ($summaryMessage)
+} elseif (-Not $failBuildOnCodeIssues) {
+    $summaryMessage = "{0} code quality issues found" -f $filteredElements.Count
```

Comment:
This should be moved to the `elseif($filteredElements.Count -eq 1` section and merge it with the current flow a la
```powershell
if ($failBuildOnCodeIssues) {
   if($filteredElements.Count -eq 1) {
      $summaryMessage = "One code quality issue found"
      Write-Output ("##vso[task.complete result=Failed;]{0}" -f $summaryMessage)
      Add-Content $summaryFilePath ($summaryMessage)
   } else {
      $summaryMessage = "{0} code quality issues found" -f $filteredElements.Count
      Write-Output ("##vso[task.complete result=Failed;]{0}" -f $summaryMessage)
      Add-Content $summaryFilePath ("**{0}** code quality issues found" -f $filteredElements.Count)
   }
} else {
   $summaryMessage = "{0} code quality issues found" -f $filteredElements.Count
   Write-Output ("##vso[task.complete result=Succeeded;]{0}" -f $summaryMessage)
   Add-Content $summaryFilePath ($summaryMessage)
}
```

Ideally we should use functions for this as well

Reply:
I moved the sections and created a function to reduce code duplication. I wasn't able to reduce it completely because of the special formatting of the summary file in the case of >1 code issues.